src/mono/System.Private.CoreLib/src/System/Runtime/Loader/AssemblyLoadContext.Mono.cs

```diff
@@ -12,6 +12,7 @@
 
 namespace System.Runtime.Loader
 {
+    [StructLayout(LayoutKind.Sequential)]
```

Comment:
I believe this construction is a little problematic and you see the symptoms of it. Just adding LayoutKind.Sequential might fix it, but depending on sequential field order in a partial class that is not even sealed, won't give you any guarantees of complete class layout. You can also see that CoreCLR depends on a different order, that is not inline with how the fields are declared, but how they are layed out by loader. I believe a better way to fix it is to get rid of `MonoManagedAssemblyLoadContext` and fix mono_alc_from_gchandle to use a cached offset to the nativeAssemblyLoadContext field and then access it based on the offset instead of depending on field order in the class.

Reply:
ok, thank you, i will try later